packages/react-aria-components/src/Menu.tsx

```diff
@@ -44,6 +45,7 @@ import {useSubmenuTrigger} from '@react-aria/menu';
 export const MenuContext = createContext<ContextValue<MenuProps<any>, HTMLDivElement>>(null);
 export const MenuStateContext = createContext<TreeState<any> | null>(null);
 export const RootMenuTriggerStateContext = createContext<RootMenuTriggerState | null>(null);
+const SelectionManagerContext = createContext<SelectionManager | null>(null);
```

Comment:
As it is now this overall approach to scoping the selection manager to each section/menu feels pretty safe. Any thoughts on if it would be helpful to expose this as well? The overall menu state context gets exported but would expose different selection manager than that of a MenuSection. I can't think of any obvious uses cases for it right now nor do we document the MenuStateContext so this is probably fine for now

Reply:
Yeah we can consider it. I guess if you wanted to build your own MenuItem component you'd need it.